python/ray/data/_internal/logical/optimizers.py

```diff
@@ -46,6 +46,12 @@ def get_logical_rules() -> List[Type[Rule]]:
     return list(_LOGICAL_RULES)
 
 
+@DeveloperAPI
+def disable_logical_rule(cls: Type[Rule]):
+    if cls in _LOGICAL_RULES:
+        _LOGICAL_RULES.remove(cls)
```

Comment:
@rickyyx i don't think that's the right approach here:

 - I'm not entirely clear on what use-cases for this would be, other than the one in testing
 - We'd be very careful in modifying global state (should be mostly immutable)